src/main/java/com/klodnicki/Bike/v2/service/BikeService.java

```diff
@@ -1,7 +1,126 @@
 package com.klodnicki.Bike.v2.service;
 
+import com.klodnicki.Bike.v2.DTO.bike.BikeForAdminResponseDTO;
+import com.klodnicki.Bike.v2.DTO.bike.BikeRequestDTO;
+import com.klodnicki.Bike.v2.model.BikeType;
+import com.klodnicki.Bike.v2.model.GpsCoordinates;
+import com.klodnicki.Bike.v2.model.entity.Bike;
+import com.klodnicki.Bike.v2.repository.BikeRepository;
+import org.modelmapper.ModelMapper;
 import org.springframework.stereotype.Service;
 
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
 @Service
-public class BikeService {
+//public class BikeService implements GenericService<BikeForAdminResponseDTO, Bike> {
+//zamienilismy BikeService dependency wstrzykiwane w controllerze na GenericBikeService
+//Żeby móc podstawić różne implementacje BikeService, nie tylko ten jeden konkretny BikeService. Jak zrobisz klasę
+//        BetterBikeService możesz bez żadnych zmian w kontrolerze z niego korzystać od razu
+public class BikeService implements GenericBikeService {
+
+    private final BikeRepository bikeRepository;
+
+    ModelMapper modelMapper = new ModelMapper();
+
+    public BikeService(BikeRepository bikeRepository) {
+        this.bikeRepository = bikeRepository;
+    }
+
+    @Override
+    public BikeForAdminResponseDTO add(BikeRequestDTO bikeRequestDTO) {
+        //using ModelMapper to convert(map) bikeRequestDTO into bike. I need Bike to be able to save it in repo
+        Bike bike = modelMapper.map(bikeRequestDTO, Bike.class);
+        Bike savedBike = bikeRepository.save(bike);
+
+        return convertBikeIntoBikeForAdminResponseDTO(savedBike); // gdyby tutaj brał bike (a nie savedBike) to by klasa
```

Comment:
Keep one solution to do something. For example here, if Model Mapper is used, just keep using it - unless there is a strong reason against it. You will develop a good practice in your coding that way and your code will also benefit from it. It will be easier to read and if you're using one way to map classes then there is only one place where it can go wrong if it's needed fixing and debugging. Otherwise you have to check and debug - like in this example - at least two places for responsibility of a bug.

Reply:
ok, thanks for that suggestion!